torch/_dynamo/guards.py

```diff
@@ -512,6 +512,13 @@ def DUPLICATE_INPUT(self, guard, source_b):
         code = [f"{ref_b} is {ref_a}"]
         self._produce_guard_code(guard, code)
 
+    def UNALIASED_INPUT(self, guard, source_b):
```

Comment:
I get why you did this downstream of `is_` but I disagree with this name - we have a fair bit of stuff in builder `wrap_tensor` ensuring we do not ever get tensors with the same input. 

I would either rename this to make it clear that this is about runtime discovery via `is_`, or write a comment and crossref it with our deduping system in builder.

Reply:
Right, maybe `UNALIASED_INPUT_NONTENSOR`?